pyro/contrib/gp/models/gplvm.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.distributions import constraints
+from torch.nn import Parameter
+
+import pyro
+from pyro.contrib.gp.util import Parameterized
+import pyro.distributions as dist
+import pyro.infer as infer
+import pyro.optim as optim
+from pyro.params import param_with_module_name
+
+
+class GPLVM(Parameterized):
+    """
+    Gaussian Process Latent Variable Model (GPLVM) model.
```

Comment:
Could you add an `Example::` section? I'm not sure how I'd use this class.

Reply:
You can find an example here: https://gist.github.com/fehiepsi/81277c39918ad4cd054157faed8b3de2

I will add an example section soon.

Reply:
This is great @fehiepsi ! Do add some of the math to the jupyter notebook if you can. This will clarify how the GPLVM is uniquely suited to a "pyronic" implementation like this one :)

Reply:
`p(X|y) \propto \prod_{d} p(y_{d} | X) p(X)` where `p(X) = N(0, Id) and p(y_{d} | X) = GP(\mu_{d}, K_{d})` (if I'm not mistaken!)

Reply:
@ysaatchi Sadly I don't have time to write a tutorial for this model. I just added an example section for this model. The gist link just showed how to use this model.